### PR TITLE
Added NULL checks in value change event & Logging transaction_uuid in msg send

### PR DESCRIPTION
**Pull request by Thanusha-D**

### CODE REVIEW

**Review by sadhyama — APPROVED** [review]

File: src/xmidtsend_rbus.c [hunk 1]
```diff
@@ -45,6 +45,10 @@ const char * contentTypeList[]={
 "application/binary"
 };
 
+void printSendMsgData(char* status, char* dest, char* transaction_uuid) {
+	ParodusInfo("Status of event: %s, destination: %s ,transation_uuid: %s\n", status?status:"NULL", dest?dest:"NULL", transaction_uuid?transaction_uuid:"NULL");
```
Comment (shilpa24balaji) [inline src/xmidtsend_rbus.c:49]: I am not sure if the ternary operator works like if(!NULL), better to explicitly add the check (status != NULL) ? status:"NULL"

File: src/xmidtsend_rbus.c [hunk 2]
```diff
@@ -430,22 +435,24 @@ void sendXmidtEventToServer(wrp_msg_t * msg, rbusMethodAsyncHandle_t asyncHandle
 				ParodusPrint("The event is having high qos retry again\n");
 				ParodusInfo("Wait till connection is Up\n");
 
-				pthread_mutex_lock(get_global_cloud_status_mut());
+			        pthread_mutex_lock(get_global_cloud_status_mut());
 				pthread_cond_wait(get_global_cloud_status_cond(), get_global_cloud_status_mut());
 				pthread_mutex_unlock(get_global_cloud_status_mut());
 				ParodusInfo("Received cloud status signal proceed to retry\n");
+                                printSendMsgData("sendUpstreamMsgToServer", notif_wrp_msg->u.event.dest, notif_wrp_msg->u.event.transaction_uuid);
```
Comment (shilpa24balaji) [inline src/xmidtsend_rbus.c:442]: We can differentiate 406 and this by adding after wait or cloud reconnect

File: src/xmidtsend_rbus.h [hunk 3]
```diff
@@ -74,6 +74,7 @@ void parseRbusInparamsToWrp(rbusObject_t inParams, char *trans_id, wrp_msg_t **e
 void createOutParamsandSendAck(wrp_msg_t *msg, rbusMethodAsyncHandle_t asyncHandle, char *errorMsg, int statuscode, rbusError_t error);
 int validateXmidtData(wrp_msg_t * eventMsg, char **errorMsg, int *statusCode);
 void xmidtQDequeue();
+void printSendMsgData(char* status, char* dest, char* transaction_uuid);
```
Comment (shilpa24balaji) [inline src/xmidtsend_rbus.h:77]: Is it better to print qos also in this?

Comment (sadhyama) [reply src/xmidtsend_rbus.h:77]: yes, better to log qos too

File: src/xmidtsend_rbus.c [hunk 4]
```diff
@@ -45,6 +45,10 @@ const char * contentTypeList[]={
 "application/binary"
 };
 
+void printSendMsgData(char* status, char* dest, int qos, char* transaction_uuid) {
+	ParodusInfo("Status of event: %s, destination: %s, qos: %d, transation_uuid: %s\n", (status!=NULL)?status:"NULL", (dest!=NULL)?dest:"NULL", qos, (transaction_uuid!=NULL)?transaction_uuid:"NULL");
```
Comment (sadhyama) [inline src/xmidtsend_rbus.c:49]: Currently log added is -
`220615-14:06:56.031367 [mod=PARODUS, lvl=INFO] [tid=10982] PARODUS: Status of event: sendUpstreamMsgToServer after cloud reconnect, destination: event:/profile-notify/MyProfile1, qos: 67, transation_uuid: d00d6632-bc24-4eb4-bba5-c1704c6317bd`

can we shorten this log a bit:
`PARODUS: status: send to server after cloud reconnect, qos: 67, dest: event:/profile-notify/MyProfile1, transaction_uuid: d00d6632-bc24-4eb4-bba5-c1704c6317bd`

Comment (Thanusha-D) [reply src/xmidtsend_rbus.c]: changes are done

**Pull request merged**